### ISSUE TITLE
Little problem with aliases

### ISSUE THREAD

**Issue by Lotzyprod** [SkriptLang/Skript#4871]
### Skript/Server Version

```markdown
2.6.2
```

### Bug Description

glow item frame and item frame entity the same

### Expected Behavior

difference between glow item frame entity and item frame entity

### Steps to Reproduce

`
on break:
  if hanging entity is item frame:
    broadcast "item frame"
  else if hanging entity is glow item frame:
    broadcast "glow item frame"
`

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by TheLimeGlass** [comment]
What version is your server and are you using paper or spigot?

**Comment by Lotzyprod** [comment]
> What version is your server and are you using paper or spigot?

paper 1.19 , but same problem and on paper 1.18.2

**Comment by TPGamesNL** [comment]
What parts don't work correctly: what happens when you click a regular item frame, and what happens when you click a glow item frame?

**Comment by AyhamAl-Ali** [comment]
Hanging entity expression returns Entity type, I don't think it can compare with itemtype

**Comment by Lotzyprod** [comment]
```
on break:
	if hanging entity is glow item frame:
		broadcast "yes"
	else:
		broadcast "no"
```
always 'no'

```
on break:
	if hanging entity is item frame:
		broadcast "yes"
	else:
		broadcast "no"
```
always yes

**Comment by Lotzyprod** [comment]
> Hanging entity expression returns Entity type, I don't think it can compare with itemtype

EntityTypes contains GLOW_ITEM_FRAME and ITEM_FRAME, its different entitytypes

**Comment by Lotzyprod** [comment]
```
on break:
	if hanging entity.getType() is EntityType.GLOW_ITEM_FRAME:
		broadcast "yes"
	else:
		broadcast "no"
```
works good

**Comment by AyhamAl-Ali** [comment]
Yep, seems like `glow item frame` is considered `item frame`, just tested

**Comment by TPGamesNL** [comment]
The reason the regular item frame condition is always true, is because GlowItemFrame (the entity) extends ItemFrame (also entity). IMO we shouldn't change this, a glow item frame is still an item frame after all. If you want to only check regular item frames, you can try `if type of hanging entity is glow item frame`.

The reason the glow item frame condition is always false, is because the list `SimpleEntityData.types` is not sorted based on inheritance: subclasses should come first in this list, because creating an instance of SimpleEntityData should select the closest type (meaning it should prefer a subclass over a superclass), but it currently just loops over the list in its registered order:
https://github.com/SkriptLang/Skript/blob/db3214d13f9a57cd4e6904d7f33bf08302b442c4/src/main/java/ch/njol/skript/entity/SimpleEntityData.java#L353-L362
Alternatively, we could hardcode this order in the registration list, but IMO it's better and safer to automate this.

This causes `item frame` entity data to be selected over `glow item frame`, causing this issue. SimpleEntityData should have a similar system to EntityData, which does sort by inheritance:
https://github.com/SkriptLang/Skript/blob/db3214d13f9a57cd4e6904d7f33bf08302b442c4/src/main/java/ch/njol/skript/entity/EntityData.java#L242-L249
